### ISSUE TITLE
Attribute-based "spell" efficacy

### ISSUE THREAD

**Issue by Turtlelord26** [trowl223/Compound_X#176]
We can differentiate between different skill levels of users of the same high-tech abilities by integrating attribute modifiers into the numbers of those abilities. I.e. "heals one wound up to X + Y*ability_mod" or "sucks out a 50% + ability_mod ~~of target's soul~~ complete copy of target's personality gestalt"

The ability modifier used is ideally the most relevant gauge of the quality of the relevant procedure, which is not necessarily the same as the most relevant gauge of the *quantity of* such procedures that particular user can perform.

From a design standpoint, I would like to see the quantity gauge (which directly determines the size of the resource pool) be the class' primary attribute and the quality gauge (which would be used as described above in spell descriptions) specifically not be the primary attribute, wherever plausible.

Such treatment for most relevant classes must await the aesthetic purge of actual magic, but specifically in the case of Medics, many Medic procedures could benefit from the integration of the user's Dex modifier.

**Comment by 1sourcecontrol** [comment]
I have huge problems with integrating too much of a class to one specific attribute. The more a class is tied to a specific attribute (or combination of attributes), the more a character _has_ to have that attribute or their character will be garbage. #177 included in this discussion.

One of the things I like a lot about compound X is the ability for me to make what I want out of a specialist or sniper or w/e. I don't have to have a lot of perception, I don't have to move fast, I can still contribute and be super strong. Knight Andréas is relatively slow at 5 (now 6) DEX, has 5 FORT, and 5 PER. It would've been a lot easier for me to push points in to PER, FORT, and DEX to be a good fighter vs putting points in STR, CHA, and INT. But CX allowed me to use other ways to try an exciting build. 
--Granting that Paladin's are built around CHA (require it for spells), I still could've been an extremely powerful fighter with a charisma of 4, if I chose to do it that way. 

Now imagine an extreme, what if all my spells had a heavy reliance on CHA, including the nanite shield? If I don't take CHA i'm basically a soldier with mediocre abilities. Might as well have taken soldier. 
Also, DnD. I really don't want compound X to be like DnD where their attributes determine their character so much that there isn't a DEX fighter, you have to have STR to be worth a darn. (I believe that's the case). Which also removes meaning from stat choice-- if you have to have certain stats, or certain stats are only good with certain classes.

Now, obviously, I'm a bit of a hypocrite for saying I have 'huge problems' with integrating too much when I proposed and PR'd the primary attribute system (at least, I think I did lol). The reason I made that design decision was so that INT would be devalued, and I tried to make the primary stat be one that a given class both represents, and a stat that balances the game well. Also, some nanite pool is typically required for play, but you don't have to max nanite pool* to build an incredible character. 
*maybe I'll need to rebalance the nanite pool gains from a primary stat to weaken the necessity of a primary stat, pending lots of feedback and thought.

Obviously you guys are much better designers than the extreme example I gave, but the extremes show where things head. I caution our travel down this road.

**Comment by Turtlelord26** [comment]
You make good points, and I very much want our producer's opinion on this rather fundamental design choice you've highlighted. I do want to mention the flip side first: that as long as we both keep our stats decently balanced at baseline and avoid letting a class rely on 2-3 attributes exclusively, we do create meaningful choices at character creation stat allocation via scarcity. I.e., if you can't have everything you want you have to pick what you don't get - and people will make different choices there and come out with different types of character.

I do admit, I have power level concerns about Str as a stat in a game that is in large part a cover-based shooter. But that's a separate issue.

**Comment by bleehu** [comment]
Firstly,  calling out the differing dimensions of Quality, vs Quantity and the implied third dimension of Rate is something I hadn't thought of before. I think that's very insightful, James! Well done!

 As for the specific design decision, I think each approach is the appropriate choice to reach differing goals. 
On the one hand, having the primary attribute be the deciding factor of the quantity gauge and having the quality be decided by something else is a fantastic guideline for creating a fun puzzle for our explorer-type players. It should be noted that almost 40% of our player base is comprised of explorers in the strict Bartle's Taxonomy. This is very good design. 
On the other, having every class have an almost equal benefit from every stat creates significantly more meaningful choices for each player and supports the stated armature of compound X back when Trowl223 was running the project: "in Compound X, you can be who you want to be." If you want to be a slow, strong gunslinger, there is a meaningful positive tactical impact from that choice. This would mean that achievers can hunt for the accomplishment of playing as a smart specialist instead of a strong one. Killers can make whatever choice they feel leads to a more satisfying body count. And while explorers will have some sacrifice in depth, I believe this design choice more than makes up for it in the breadth of incomparables for characters which are now not only viable but advantageous to play. This is great design. 
TL;DR: I would like to decentralize value of a given stat point as much as possible. In so doing I believe we pose more meaningful questions to our players and make the game more fun in the long run for different player types.

**Comment by Turtlelord26** [comment]
We touched briefly on the idea of moving Nanites to a full secondary stat yesterday evening. On reflection, I rather like the idea given our design goals. Enabling the Tanky low-Nanite caster would be new and different - and functionally make Paladin just a different stat allocation on whatever hightech class we give healing to, which I think is awesome.

What do you two think of tying Nanites to Per and Int?, on the theory that having lots of Int lets you manage more nanites and Per lets you allocate the ones you have more effectively (which is equivalent to having more) because you can sense where they would be most effective. Design-wise, Per is the only non-Luck stat not tied to a secondary at the moment, and Int seems weak to me as-is because you can pump Cha, dump Int and still have lots of skills.

**Comment by bleehu** [comment]
It's hard for me to at a glance get a feel for the relative balance of Per, Int and for the ground state, say, Strength. Let's put a pin in this idea (maybe close the issue?) until we've run a few builds in C2.0 to actually test the balance of the new Stat system?

**Comment by 1sourcecontrol** [comment]
At the bottom of the sheet "other-changelist" in CX_WepArm_Analysis, you'll find something like this:
![image](https://user-images.githubusercontent.com/2192156/32349210-7c563058-bfd3-11e7-956c-a5b90cfd14c5.png)


I used this little value assignment system to try and balance the primary stats and come up with solutions. I updated it for PER and INT deciding Nanite pool sizes. It needs some work, but it gives a general idea of how things may be working.

I think it would be reasonable to use Per and Int. I really like PER, but I'm concerned with Int having Nanites, bonus feats, and skill points. CHA would probably be dumped if you max INT (unless you wanted to be tanky with CHA and FORT). But dumping PER would be a strategic choice. Having PER with nanites tho, would allow a mage to both be high in Nanites and weapon accuracy (of course, they only have pistols so it doesn't really matter). 

Anywhos, I am leaning towards wanting to make the change that you @Turtlelord26 suggested. 
At the very least, I need to go through and check the class primary stats for imbalance, but I have a feeling that having DEX, STR, or FORT tied to nanites would be a bad move (and we already have that in C2.0).

**Comment by 1sourcecontrol** [comment]
Actually, I totally forgot that if a primary stat has 2 secondary stats, you have to choose which one to put the points in to. I think that pretty much solves the dilemma for INT, since you can Max CHA and INT to get skill points and nanites. Or max PER and INT or PER and CHA.

**Comment by 1sourcecontrol** [comment]
@bleehu, what do you think about my reply?

**Comment by bleehu** [comment]
I appreciate the insight, but until we test, I don't know how well your model matches the reality of play.

**Comment by 1sourcecontrol** [comment]
So you're saying, forget it until we try it? It's not just about the model, although I made that model based on pre combat 2.0 play. I feel like we can easily do some tests, but you don't think there's any problem with having Nanites linked to a primary stat?

**Comment by bleehu** [comment]
Primary stat as opposed to secondary stat?

Yes, I strongly suggest before we continue to make tweaks that we run a full test.

**Comment by 1sourcecontrol** [comment]
Yeah, primary as opposed to secondary.

What constitutes a full test? Several missions with the new stat system? I feel like all we need is to have a couple people make characters. If we measure their decision making process then we can try and find balance

**Comment by 1sourcecontrol** [comment]
Well, I guess measuring their decision making maybe isn't a perfect way to go about it. But I'm open to ideas.

**Comment by bleehu** [comment]
I would say two missions with different players. It's not enough for me to wrap my head around the way the "rubber meets the road" until I actually see it happen.

**Comment by 1sourcecontrol** [comment]
I feel like we're talking about two different things here, what is it that you want to test with two missions?


What I'm talking about is the extra value added by making each class have a 'Class Primary Attribute'. My theory is that this extra value can lead to imbalance when that Class Primary Attribute is Dex, Str, or Fort (or almost any other). We don't need missions to test this, all we need is people to make a bunch of C2.0 characters. 

If 10 people make characters (and the 10 characters all have their Class Primary Attribute maxed as high as possible) and if I ask them _"why did you do that?"_ and they say _"This _____ Class Primary Attribute gets me the most bang for my buck!"_ Then I've completely failed in balancing the way I've hoped to; like you said earlier: 

>TL;DR: I would like to decentralize value of a given stat point as much as possible. 

Now, if a person isn't trying to build a character based on what the stats bring, then that's not a problem. But for the other types of players, this example would be a huge red flag for me because it looks like I've reduced the number of meaningful choices.

**Comment by 1sourcecontrol** [comment]
Alrighty, so @bleehu discussed this after I wrote the post.

My vote is for a conversion for Nanites as a Secondary stat under PER and INT.
If anyone comes up with another argument for balance on either the Nanites as a secondary stat vs Nanites as a class primary stat, lemee know. 

Regardless, things need to get tested for us to see the balance.

Leaving this issue open until we can test and arrive at some conclusion.

**Comment by Turtlelord26** [comment]
This is very stale and I am calling it dead, the sole remaining outstanding issue of nanite pool sizes having been subsumed by the 'Nanites as a Resource' design discussion.
